src/App.js

```diff
@@ -7,23 +7,46 @@ import React, { useEffect, useState } from 'react'
 function App() {
   const [movies, setMovies] = useState(movieData);
   const [selectedMovie, setSelectedMovie] = useState(null)
+  const [error, setError] = useState('')
+  const backButton = () => setSelectedMovie(null)
 
- const fetchData = () => {
+  const fetchData = () => {
     fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
-    .then(resp => resp.json())
-    .then(data => setMovies(data))
- }
+      .then(response => {
+        if (!response.ok) {
+          throw new Error('Failed to fetch all movies');
```

Comment:
How does this thrown error get surfaced to the user? It might be a better user experience to hold on to this error in state, and then conditionally render a helpful message to the user based on that state!!